### PR TITLE
Add validation of the number of distributor stripes from config and a…

### PR DESCRIPTION
**Pull request by geirst**
…dd more asserts.

This ensures the number of stripes is a power of 2 and within MaxStripes boundary.

@vekterli please review
@toregge FYI

### CODE REVIEW

File: storage/src/vespa/storage/common/bucket_stripe_utils.h [hunk 1]
```diff
@@ -22,7 +22,14 @@ size_t stripe_of_bucket_key(uint64_t key, uint8_t n_stripe_bits) noexcept;
...
  */
-uint8_t calc_num_stripe_bits(size_t n_stripes) noexcept;
+uint8_t calc_num_stripe_bits(uint32_t n_stripes) noexcept;
+
+/**
+ * Validates the number of stripes and returns the (potentially adjusted) value.
+ *
+ * This ensures the number of stripes is a power of 2 and within MaxStripes boundary.
+ */
+uint32_t validate_num_stripes(uint32_t n_stripes) noexcept;
```
Comment (vekterli) [inline storage/src/vespa/storage/common/bucket_stripe_utils.h:32]: Consider adding `[[nodiscard]]` attribute

Comment (vekterli) [inline storage/src/vespa/storage/common/bucket_stripe_utils.h:32]: To me, the prefix "validate" tends to imply that the function performs some explicit action if the validation fails (such as throwing an exception). This function communicates that this is not the case by being tagged `noexcept`, but I wonder if there's a more fitting name for it still. Perhaps just `adjusted_num_stripes` ref. the comment?

File: storageserver/src/vespa/storageserver/app/distributorprocess.cpp [hunk 2]
```diff
@@ -32,6 +33,21 @@ DistributorProcess::shutdown()
     _node.reset();
 }
 
+namespace {
+
+uint32_t
+validate_num_distributor_stripes(uint32_t cfg_n_stripes)
```
Comment (vekterli) [inline storageserver/src/vespa/storageserver/app/distributorprocess.cpp:39]: Consider if function should be slightly renamed here as well

**Review by vekterli — APPROVED** [review]
:+1:

File: storage/src/vespa/storage/common/bucket_stripe_utils.h (hunk 1 above)
Comment (geirst) [reply storage/src/vespa/storage/common/bucket_stripe_utils.h]: Fixed

Comment (geirst) [reply storage/src/vespa/storage/common/bucket_stripe_utils.h]: I agree. Fixed.

File: storageserver/src/vespa/storageserver/app/distributorprocess.cpp (hunk 2 above)
Comment (geirst) [reply storageserver/src/vespa/storageserver/app/distributorprocess.cpp]: Fixed

**Comment by geirst** [comment]
@vekterli PTAL

**Review by vekterli — APPROVED** [review]
🦄👍

**Pull request merged**